verifier/src/x509.rs

```diff
@@ -29,6 +35,53 @@ impl From<mbedtls::Error> for Error {
     }
 }
 
+impl From<der::Error> for Error {
+    fn from(src: der::Error) -> Self {
+        Error::Der(src)
+    }
+}
+
+impl From<Error> for CertificateChainVerifierError {
+    fn from(error: Error) -> Self {
+        match error {
+            // Any error (expired, revoked, etc) in certificate chain verification comes back as
+            // `X509CertVerifyFailed`
+            Error::MbedTls(mbedtls::Error::X509CertVerifyFailed) => {
+                CertificateChainVerifierError::SignatureVerification
+            }
+            _ => CertificateChainVerifierError::GeneralCertificateError,
+        }
+    }
+}
+
+/// A certificate chain verifier that uses MbedTls as the backend
+#[derive(Debug)]
+pub struct X509CertificateChainVerifier {
```

Comment:
not fond of this name, seems too wordy and possibly ambiguous with the `x509_cert` crate. Maybe mbedtls should be in the name, but that feels like it's exposing the backend that is meant to be hidden, other than mbedtls seems to be a PIA for the different SDK platforms.

Reply:
With the risk of bikeshedding, I think I would've chosen `MbedtlsCertificateChainVerifier` (named a feature that brings in `mbedtls` in the crate), and the file probably `mbedtls.rs` or something like that. The thinking behind that the trait `CertificateChainVerifier` already implies these are X509 certs, given that one of the arguments of its only method accepts a list of 509 certificates. So it doesn't seem like there's a world in which `CertificateChainVerifier` verifies any other kind of certificate, so the implementation is really just selecting which backend one wants to use. And as such I would've named the backend implementation that uses `mbedtls` `MbedtlsCertificateChainVerifier`.

Reply:
I will update to that, I wasn't really fond of the x509 naming, and I agree "certificate chain" => x509

Reply:
updated now